docs/advanced-topics/wes/wes-agc-tutorial.rst

```diff
@@ -0,0 +1,332 @@
+                "end_time" : "2022-03-04T17:23:17.196Z",
+                "stdout" : "s3://agc-example123-us-west-2/project/dockstoreAgcTutorialProject/userid/userM2QLG/context/ctx1/cromwell-execution/w/8e8e9f4b-fb1a-41df-bc37-9396d6f97db5/call-hello/hello-stdout.log",
+                "stderr" : "s3://agc-example123-us-west-2/project/dockstoreAgcTutorialProject/userid/userM2QLG/context/ctx1/cromwell-execution/w/8e8e9f4b-fb1a-41df-bc37-9396d6f97db5/call-hello/hello-stderr.log",
+                "exit_code" : 0
+              } ],
+              "outputs" : {
+                "id" : "8e8e9f4b-fb1a-41df-bc37-9396d6f97db5",
+                "outputs" : {
+                  "w.hello.out" : "Hello from AGC"
+                }
+              }
+            }
+
+    Notice that the output for task ``hello`` of workflow ``w`` is "Hello from AGC".
+
+FastQ Read Counts Workflow
+--------------------------
+The Dockstore entry associated with this workflow can be found here `agc-fastq-read-counts <https://dockstore.org/workflows/github.com/dockstore-testing/wes-testing/agc-fastq-read-counts:main?tab=info>`_.
+
+This WDL workflow tabulates read counts of the input fastq file.
+
+    *Dockstore.wdl*
+
+    .. code:: text
+
+        version 1.0
+
+        workflow fastqReadCounts {
+
+            call countFastqReads
+
+            output {
+                File totalReadsFile = countFastqReads.totalReadsFile
+            }
+        }
+
+
+
+        task countFastqReads {
+
+            input {
+                Array[File] inputFastq
+
+                Int memSizeGB = 4
+                Int diskSizeGB = 128
+                String dockerImage = "biocontainers/samtools:v1.9-4-deb_cv1"
+            }
+
+            command <<<
+
+                set -o pipefail
+                set -e
+                set -u
+                set -o xtrace
+
+                READ_COUNT=0
+
+                for fq in ~{sep=' ' inputFastq}
+                do
+                      FILE_COUNT=$(zcat "${fq}" | wc -l )/4
+                      READ_COUNT=$(( $READ_COUNT + $FILE_COUNT ))
+                done
+
+                echo $READ_COUNT > total_reads.txt
+            >>>
+
+            output {
+
+                File totalReadsFile  = "total_reads.txt"
+            }
+
+            runtime {
+                memory: memSizeGB + " GB"
+                disks: "local-disk " + diskSizeGB + " SSD"
+                docker: dockerImage
+                preemptible: 1
+            }
+        }
+
+1. This workflow takes an array of files as an input. Create a file named ``input.json`` in your working directory with contents:
+
+    *input.json*
+
+    .. code:: text
+
+        {
+            "fastqReadCounts.countFastqReads.inputFastq": ["s3://human-pangenomics/working/HPRC_PLUS/HG005/raw_data/Illumina/child/5A1-24481579/5A1_S5_L001_R1_001.fastq.gz"]
+        }
+
+2. As a requirement of AGC input parsing, create a second file named ``agcInputs.json`` in your working directory.
```

Comment:
This is a little confusing, due to the AGC implementation -- you have the text "input" 5 times in this and the next sentence, so it's hard to track which "input" text corresponds to what. :) Also, on line 279 you introduce the  `-a` switch without being explicit about it.

How about changing the name of agcInputs.json to agcWrapper.json? And then something like this:

As a requirement for AGC requests, create a second file named `agcWrapper.json`... This file specifies the input JSON for the...

On line 275, "by passing the Dockstore CLI the entry name, the AGC wrapper, and the input JSON. The input JSON is specified with the ``--attach` or -`a` switch.

A bit rough, and just an idea. I'm not sure if renaming the AGC file to wrapper is consistent with their doc, and if it's not, scratch this idea.